### PR TITLE
JDK-8301076: Replace NULL with nullptr in share/prims/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/prims/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301076](https://bugs.openjdk.org/browse/JDK-8301076): Replace NULL with nullptr in share/prims/

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to [b1ff44d7](https://git.openjdk.org/jdk/pull/12188/files/b1ff44d75fd0ba481ab27e99a3a0f03c09cbf241)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [b1ff44d7](https://git.openjdk.org/jdk/pull/12188/files/b1ff44d75fd0ba481ab27e99a3a0f03c09cbf241)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12188/head:pull/12188` \
`$ git checkout pull/12188`

Update a local copy of the PR: \
`$ git checkout pull/12188` \
`$ git pull https://git.openjdk.org/jdk pull/12188/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12188`

View PR using the GUI difftool: \
`$ git pr show -t 12188`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12188.diff">https://git.openjdk.org/jdk/pull/12188.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/prims/jni.cpp [hunk 1]
```diff
@@ -2226,16 +2226,16 @@ JNI_END
...
  HOTSPOT_JNI_GETSTRINGUTFCHARS_ENTRY(env, string, (uintptr_t *) isCopy);
-  char* result = NULL;
+  char* result = nullptr;
   oop java_string = JNIHandles::resolve_non_null(string);
   typeArrayOop s_value = java_lang_String::value(java_string);
-  if (s_value != NULL) {
+  if (s_value != nullptr) {
     size_t length = java_lang_String::utf8_length(java_string, s_value);
-    /* JNI Specification states return NULL on OOM */
+    /* JNI Specification states return nullptr on OOM */
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jni.cpp:2234]: null

File: src/hotspot/share/prims/jni.cpp [hunk 2]
```diff
@@ -2414,17 +2414,17 @@ JNI_ENTRY_NO_PRESERVE(ElementType*, \
   ElementType* result; \
   int len = a->length(); \
   if (len == 0) { \
-    if (isCopy != NULL) { \
+    if (isCopy != nullptr) { \
       *isCopy = JNI_FALSE; \
     } \
-    /* Empty array: legal but useless, can't return NULL. \
+    /* Empty array: legal but useless, can't return nullptr. \
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jni.cpp:2420]: null

File: src/hotspot/share/prims/jni.cpp [hunk 3]
```diff
@@ -2414,17 +2414,17 @@ JNI_ENTRY_NO_PRESERVE(ElementType*, \
...
      * Return a pointer to something useless. \
      * Avoid asserts in typeArrayOop. */ \
     result = (ElementType*)get_bad_address(); \
   } else { \
-    /* JNI Specification states return NULL on OOM */                    \
+    /* JNI Specification states return nullptr on OOM */                    \
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jni.cpp:2425]: null

File: src/hotspot/share/prims/jni.cpp [hunk 4]
```diff
@@ -2830,20 +2830,20 @@ JNI_ENTRY(const jchar*, jni_GetStringCritical(JNIEnv *env, jstring string, jbool
...
-    if (isCopy != NULL) *isCopy = JNI_FALSE;
+    if (isCopy != nullptr) *isCopy = JNI_FALSE;
   } else {
     // Inflate latin1 encoded string to UTF16
     typeArrayOop s_value = java_lang_String::value(s);
     int s_len = java_lang_String::length(s, s_value);
     ret = NEW_C_HEAP_ARRAY_RETURN_NULL(jchar, s_len + 1, mtInternal);  // add one for zero termination
-    /* JNI Specification states return NULL on OOM */
-    if (ret != NULL) {
+    /* JNI Specification states return nullptr on OOM */
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jni.cpp:2839]: null

File: src/hotspot/share/prims/jniCheck.cpp [hunk 5]
```diff
@@ -92,7 +92,7 @@ static struct JNINativeInterface_ * unchecked_jni_NativeInterface;
 extern "C" {                                                             \
   result_type JNICALL header {                                           \
     Thread* cur = Thread::current_or_null();                             \
-    if (cur == NULL || !cur->is_Java_thread()) {                         \
+    if (cur == nullptr || !cur->is_Java_thread()) {                         \
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jniCheck.cpp:95]: align

File: src/hotspot/share/prims/jniCheck.cpp [hunk 6]
```diff
@@ -932,7 +932,7 @@ JNI_ENTRY_CHECKED(ResultType,  \
     functionEnter(thr); \
     va_list args; \
     IN_VM( \
-      jniCheck::validate_call(thr, NULL, methodID, obj); \
+      jniCheck::validate_call(thr, nullptr, methodID, obj); \
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jniCheck.cpp:935]: align

File: src/hotspot/share/prims/jniCheck.cpp [hunk 7]
```diff
@@ -950,7 +950,7 @@ JNI_ENTRY_CHECKED(ResultType,  \
                                     va_list args)) \
     functionEnter(thr); \
     IN_VM(\
-      jniCheck::validate_call(thr, NULL, methodID, obj); \
+      jniCheck::validate_call(thr, nullptr, methodID, obj); \
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jniCheck.cpp:953]: align

File: src/hotspot/share/prims/jniCheck.cpp [hunk 8]
```diff
@@ -966,7 +966,7 @@ JNI_ENTRY_CHECKED(ResultType,  \
                                     const jvalue * args)) \
     functionEnter(thr); \
     IN_VM( \
-      jniCheck::validate_call(thr, NULL, methodID, obj); \
+      jniCheck::validate_call(thr, nullptr, methodID, obj); \
```
Comment (jdksjolen) [inline src/hotspot/share/prims/jniCheck.cpp:969]: align

File: src/hotspot/share/prims/jniCheck.cpp (hunk 6 above)
Comment (jdksjolen) [reply src/hotspot/share/prims/jniCheck.cpp:935]: Not aligned from start, not fixing.

File: src/hotspot/share/prims/jniCheck.cpp (hunk 7 above)
Comment (jdksjolen) [reply src/hotspot/share/prims/jniCheck.cpp:953]: Not aligned from start, not fixing.

File: src/hotspot/share/prims/jniCheck.cpp (hunk 8 above)
Comment (jdksjolen) [reply src/hotspot/share/prims/jniCheck.cpp:969]: Not aligned from start, not fixing.

**Comment by jdksjolen** [comment]
Passes tier1.

File: src/hotspot/share/prims/jvmtiImpl.hpp [hunk 9]
```diff
@@ -549,7 +549,7 @@ class JvmtiDeferredEventQueue : public CHeapObj<mtInternal> {
   void oops_do(OopClosure* f, CodeBlobClosure* cf) NOT_JVMTI_RETURN;
 };
 
-// Utility macro that checks for NULL pointers:
-#define NULL_CHECK(X, Y) if ((X) == NULL) { return (Y); }
+// Utility macro that checks for null pointers:
+#NULL nullptr_CHECK(X, Y) if ((X) == nullptr) { return (Y); }
```
Comment (alexmenkov) [inline src/hotspot/share/prims/jvmtiImpl.hpp:553]: causes compilation error

File: src/hotspot/share/prims/methodHandles.cpp [hunk 10]
```diff
@@ -1043,7 +1043,7 @@ static bool advertise_con_value(int which) {
 
 #undef ONE_PLUS
 #undef VALUE_COMMA
-#undef STRING_NULL
+#undef STRING_nullptr
```
Comment (kimbarrett) [inline src/hotspot/share/prims/methodHandles.cpp:1046]: Should not be changed.

File: src/hotspot/share/prims/methodHandles.cpp [hunk 11]
```diff
@@ -1021,8 +1021,8 @@ void MethodHandles::trace_method_handle_interpreter_entry(MacroAssembler* _masm,
 static const int con_value_count = EACH_NAMED_CON(ONE_PLUS, IGNORE_REQ) 0;
 #define VALUE_COMMA(scope,value) scope::value,
 static const int con_values[con_value_count+1] = { EACH_NAMED_CON(VALUE_COMMA, IGNORE_REQ) 0 };
-#define STRING_NULL(scope,value) #value "\0"
-static const char con_names[] = { EACH_NAMED_CON(STRING_NULL, IGNORE_REQ) };
+#define STRING_nullptr(scope,value) #value "\0"
```
Comment (kimbarrett) [inline src/hotspot/share/prims/methodHandles.cpp:1024]: Should not be changed.

File: src/hotspot/share/prims/methodHandles.cpp [hunk 12]
```diff
@@ -1021,8 +1021,8 @@ void MethodHandles::trace_method_handle_interpreter_entry(MacroAssembler* _masm,
...
+static const char con_names[] = { EACH_NAMED_CON(STRING_nullptr, IGNORE_REQ) };
```
Comment (kimbarrett) [inline src/hotspot/share/prims/methodHandles.cpp:1025]: Should not be changed.

File: src/hotspot/share/prims/upcallLinker.cpp [hunk 13]
```diff
@@ -97,7 +97,7 @@ JavaThread* UpcallLinker::on_entry(UpcallStub::FrameData* context) {
   context->old_handles = thread->active_handles();
 
   // For the profiler, the last_Java_frame information in thread must always be in
-  // legal state. We have no last Java frame if last_Java_sp == NULL so
+  // legal state. We have no last Java frame if last_Java_sp == null so
```
Comment (kimbarrett) [inline src/hotspot/share/prims/upcallLinker.cpp:100]: s/null/nullptr/ ?

**Review by kimbarrett — CHANGES_REQUESTED** [review]

**Comment by jdksjolen** [comment]
Thanks for these review comments, I've fixed your suggestions. I must've accidentally submitted a different branch for tier1 testing, as this doesn't compile.

**Review by dholmes-ora — APPROVED** [review]
A few adjustments but overall looks good. Thanks.

**Comment by jdksjolen** [comment]
All tests passing in GHA. @alexmenkov , would you mind approving :)?

**Comment by jdksjolen** [comment]
Tier1 is passing and this has 2 approvals, I'm integrating. Thank you for the reviews.

/integrate

**Pull request closed without merging**